### ISSUE TITLE
Using custom fields added via Settings Template resets the Templates paths on a Report

### ISSUE THREAD

**Issue by adrianbunea** [barrelstrength/sprout#242]
### Description
I have added two Date fields to my Report, after saving the paths for the first time, the two new fields appear, after filling in any of those fields and save, I am returned to the first field, where I need to input the Template paths, and the two custom fields are gone.

Paths are setup like this:
![image](https://user-images.githubusercontent.com/42218790/138892563-958001e8-eb2d-4eac-8dfa-a7c3e9dfa485.png)

Files are setup like this:
![image](https://user-images.githubusercontent.com/42218790/138892625-12be4ad0-b1d1-480f-8792-0f257d520257.png)

Settings template is setup like this:
```
{% import "_includes/forms" as forms %}

{{ forms.dateField({
  label: "Start Date"|t,
  instructions: "The date when to start looking for payments."|t,
  id: "datetimeStartDate",
  name: "datetimeStartDate",
  value: (settings.datetimeStartDate is defined ? settings.datetimeStartDate : null),
  required: false
}) }}


{{ forms.dateField({
  label: "End Date"|t("plugin-handle"),
  instructions: "The date when to stop looking for payments."|t,
  id: "datetimeEndDate",
  name: "datetimeEndDate",
  value: (settings.datetimeEndDate is defined ? settings.datetimeEndDate : null),
  required: false
}) }}

{%- includejs %}

    $('#settings-datetimeStartDate-field .datewrapper input:first-of-type').datepicker($.extend({
    defaultDate: new Date({{ now.year }}, {{ now.month }}, {{ now.day }})
    }, Craft.datepickerOptions));

    $('#settings-datetimeEndDate-field .datewrapper input:first-of-type').datepicker($.extend({
    defaultDate: new Date({{ now.year }}, {{ now.month }}, {{ now.day }})
    }, Craft.datepickerOptions));

    var $timePickerStartDate = $('#settings-datetimeStartDate-field .timewrapper input:first-of-type');
    var $timePickerEndDate = $('#settings-datetimeEndDate-field .timewrapper input:first-of-type');

    $timePickerStartDate.timepicker($.extend({}, Craft.timepickerOptions));
    $timePickerEndDate.timepicker($.extend({}, Craft.timepickerOptions));

{%- endincludejs %}
```

After saving the report template paths, I get this:
![image](https://user-images.githubusercontent.com/42218790/138893849-e30bd23b-848e-44c0-b31a-3b8c105f5745.png)

After inputting the two dates, and save it, I get returned to the original screen where I need to add the template paths:
![image](https://user-images.githubusercontent.com/42218790/138893998-df517d83-01ca-4a6b-97dc-50441bc26ebd.png)

### Steps to reproduce
1. Create a new Report
2. Setup paths for templates, especially for a settings, and make sure to add a date field
3. Save the Report
4. Edit the Report again
5. Modify the custom date field
6. Save the template again => Get returned to the first view, with the template paths gone and no more fields.

### Additional info

- Craft version: Craft Pro 3.5.8
- PHP version: 7.2.34
- Database driver & version: MySQL 5.7.31
- Plugins & versions:
    Sprout Email | 4.4.8
    Sprout Reports | 1.5.9

**Comment by BenParizek** [comment]
Thanks for the clear description @adrianbunea. I've just released Sprout Reports v1.5.12 which should solve the issue.

It seems Craft must have changed how the Timepicker was loading which was causing a javascript error and disabling those autosuggest template fields during the second submission.

**Comment by paulgoodfield** [comment]
I'm getting a similar issue - when I run a report and then click on the 'cog' icon at the top right of the screen to show my custom settings date field, if I alter the field value and then click the 'Run Report' button again I get the following error:

```
TypeError: Argument 1 passed to craft\web\View::renderTemplate() must be of the type string, null given, called in /app/craft/vendor/barrelstrength/sprout-base-reports/src/datasources/CustomTwigTemplate.php on line 215 and defined in /app/craft/vendor/craftcms/cms/src/web/View.php:388
Stack trace:
#0 /app/craft/vendor/barrelstrength/sprout-base-reports/src/datasources/CustomTwigTemplate.php(215): craft\web\View->renderTemplate(NULL, Array)
barrelstrength/sprout-reports#1 /app/craft/vendor/barrelstrength/sprout-base-reports/src/datasources/CustomTwigTemplate.php(52): barrelstrength\sproutbasereports\datasources\CustomTwigTemplate->processFrontEndResultsTemplate(Object(barrelstrength\sproutbasereports\elements\Report), Array)
barrelstrength/sprout-reports#2 /app/craft/vendor/barrelstrength/sprout-base-reports/src/controllers/ReportsController.php(149): barrelstrength\sproutbasereports\datasources\CustomTwigTemplate->getDefaultLabels(Object(barrelstrength\sproutbasereports\elements\Report))
barrelstrength/sprout-reports#3 [internal function]: barrelstrength\sproutbasereports\controllers\ReportsController->actionResultsIndexTemplate(Object(barrelstrength\sproutbasereports\elements\Report), 238665)
barrelstrength/sprout-reports#4 /app/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
barrelstrength/sprout-reports#5 /app/craft/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
barrelstrength/sprout-reports#6 /app/craft/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('results-index-t...', Array)
barrelstrength/sprout#10 /app/craft/vendor/craftcms/cms/src/web/Application.php(293): yii\base\Module->runAction('sprout-base-rep...', Array)
barrelstrength/sprout-reports#8 /app/craft/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('sprout-base-rep...', Array)
barrelstrength/sprout-reports#9 /app/craft/vendor/craftcms/cms/src/web/Application.php(278): yii\web\Application->handleRequest(Object(craft\web\Request))
barrelstrength/sprout-reports#10 /app/craft/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
barrelstrength/sprout-reports#11 /app/craft/web/index.php(21): yii\base\Application->run()
barrelstrength/sprout-reports#12 {main}
```

When I go back to edit the report, the template paths for both the results and settings twig files are now empty.

Craft 3.7.34

Plugin List:

AsyncQueue	2.3.0
Calendar	3.3.8
Commerce Insights	1.0.11
Commerce Widgets	2.0.21
Cookies	1.1.15
CP Field Inspect	1.2.6
Craft Commerce	3.4.11
Gift Voucher	2.7.0
Incognito Field	1.2.0
Patrol	3.1.3
Redactor	2.10.5
SEO	3.7.4
Sprout Reports	1.5.14
Super Table	2.7.1
